tests/testthat/test-class.R

```diff
@@ -183,3 +183,8 @@ describe("default constructor", {
     expect_s3_class(x@x, "foo1")
   })
 })
+
+test_that("c(<R7_class>, ...) gives error", {
+  foo1 <- new_class("foo1")
+  expect_error(c(foo1, foo1))
```

Comment:
This would be a good place to use `expect_snapshot(..., error = TRUE)` because that make it easier to verify the text of the error is useful (not that important here since there's no interpolation, but is a good general pattern, IMO).